src/main.rs

```diff
@@ -189,47 +191,68 @@ fn main() {
-                    r
-                });
+            let spawn_fut = futures::future::lazy(move |_cx: &mut Context| async move {
+                setup_future.await;
+
+                let tls_socket = acceptor
+                    .accept(tcp_stream)
+                    .await
+                    .map_err(|err| {
+                        task_warn!("TLS handshake failed"; "error" => format!("{}", err));
+                        io::Error::new(io::ErrorKind::Other, err);
+                    })
+                    .unwrap();
+
+                let mut bridge =
+                    bridge::Bridge::create(cloned_url, tls_socket, irc_server_name, ctx)
+                        .await
+                        .unwrap();
+
+                // TODO: I belive this is what taskedfutures did (except it was previously spawned
+                // off inside Bridge::create). I will come back and make sure this is correct
+                // later.
+                loop {
+                    if let Err(_) = bridge.poll_irc().await {
+                        break;
+                        // TODO: log this
+                    }
+                    if let Err(_) = bridge.poll_matrix().await {
+                        break;
+                        // TODO: log this
+                    }
+                }
+
+                task_info!("Finished");
+            });
 
             // We spawn the future off, otherwise we'd block the stream of incoming connections.
             // This is what causes the future to be in its own chain.
-            handle.spawn(future);
+            //
+            // These are spawn_local due to it not requiring spawn_fut to be Send, but it could
+            // possibly be `spawn` in the future after changing trait bounds.
+            tokio::task::spawn_local(spawn_fut);
         } else {
             // Same as above except with less TLS.
-            let future = setup_future
-                .and_then(move |socket| {
-                    bridge::Bridge::create(new_handle, cloned_url, socket, irc_server_name, ctx)
-                })
-                .map(|bridge| bridge.into_future())
-                .flatten()
-                .map_err(|err| task_warn!("Unhandled IO error"; "error" => format!("{}", err)));
-
-            handle.spawn(future);
-        };
+            let spawn_fut = futures::future::lazy(move |_cx: &mut Context| async move {
+                setup_future.await;
+
+                let mut bridge =
+                    bridge::Bridge::create(cloned_url, tcp_stream, irc_server_name, ctx)
+                        .await
+                        .unwrap();
+                loop {
+                    if let Err(_) = bridge.poll_irc().await {
+                        break;
+                        // TODO: log this
+                    }
+                    if let Err(_) = bridge.poll_matrix().await {
+                        break;
+                        // TODO: log this
```

Comment:
Just add simple logs here, then it's done :)

Reply:
These logs appear on my branch https://github.com/VanillaBrooks/matrix-ircd/blob/async-await/src/main.rs#L231-L247. more logging is in the tls bridge version above.